docs/RELEASING.md

```diff
@@ -0,0 +1,55 @@
+# Releasing urfave/gimme
+
+Releasing small batches often is [backed by
+research](https://itrevolution.com/accelerate-book/) as part of the
+virtuous cycles that keep teams and products healthy.
+
+To that end, the overall goal of the release process is to send
+changes out into the world as close to the time the commits were
+merged to the `main` branch as possible. In this way, the community
+of humans depending on this library are able to make use of the
+changes they need **quickly**, which means they shouldn't have to
+maintain long-lived forks of the project, which means they can get
+back to focusing on the work on which they want to focus. This also
+means that the @urfave/gimme team should be able to focus on
+delivering a steadily improving product with significantly eased
+ability to associate bugs and regressions with specific releases.
+
+## Process
+
+- Release versions follow [semantic versioning](https://semver.org/)
+- Releases are associated with **signed, annotated git tags**[^1].
+- Release notes are **automatically generated**[^1].
+
+In the `main` or `v2-maint` branch, the current version is always
+available via:
+
+```sh
+git describe --always --dirty --tags
+```
+
+**NOTE**: if the version reported contains `-dirty`, this is
+indicative of a "dirty" work tree, which is not a great state for
+creating a new release tag. Seek help from @urfave/gimme teammates.
+
+For example, given a described version of `v1.7.1-7-g65c7203` and a
+diff of `v1.7.1...` that contains only bug fixes, the next version
+should be `v1.7.2`:
+
+- Update `GIMME_VERSION` in [gimme](https://github.com/urfave/gimme/blob/main/gimme) via pull request.
+
+- Once that is approved and merged make a tag locally:
+```sh
+TAG_VERSION=v1.7.2 make tag
+git push origin v1.7.2
+```
```

Comment:
I'm not likely to be making releases, so this is probably a very moot point that I'm happy to concede on, but wouldn't it be the same / sufficient to make the tag via the web UI?

(I haven't set up commit/tag signing locally, for example, so web UI actions are the only way I'll have signed objects.  That being said I don't actually use `gimme` any more and haven't for a long time; I'm just still here because @meatballhat is one of the best humans there ever was and I'm happy to lend a hand if I can since I helped create the thing eons ago. :smile:)

Reply:
v1.5.1 was signed that way, by crating the tag on the web view, and it does create a signed tag. v1.5.2 isn't signed at all. So I have no objections to the tags being signed by github.

However the `tag` target in the Makefile refuses to make the tag unless you've updated the version number in source, which we forgot to do for v1.7.0. Presumably because @meatballhat made the tag using git locally. I think let's do our best to protect ourselves and keep this method.